### DISCUSSION TITLE
validation on sqs listener

### DISCUSSION THREAD

**Discussion by nekperu15739** [awspring/spring-cloud-aws#532] [Q&A]
Hi, im trying to validate the payload on a method with `@SqsListener`, however checking underhood looks like there is not an easy why to do it having in mind that `PayloadArgumentResolver` by default use a `NoOpValidator`. **could you share a way to do set a validator?** 

NOTE:
similar question was asked here: [stackoverflow](https://stackoverflow.com/questions/51438961/how-to-get-javax-validation-payload-validation-for-spring-cloud-aws-in-queuemess) however, those workarounds are not good enough because lose other functionalities.

**Comment by maciejwalkowiak** [comment]
@nekperu15739 which Spring Cloud AWS version?

**Reply by tomazfernandes** [reply]
From the SO issue, seems like 2.x.

3.0 has payload validation but TBH I haven’t given much thought to it yet, so while it should work it’s lacking more comprehensive testing.

**Reply by nekperu15739** [reply]
> @nekperu15739 which Spring Cloud AWS version?

2.4.2
